ob1k-cache/src/main/java/com/outbrain/ob1k/cache/SimpleLocalAsyncCache.java

```diff
@@ -0,0 +1,104 @@
+package com.outbrain.ob1k.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.ExecutionError;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.outbrain.ob1k.concurrent.ComposableFuture;
+import com.outbrain.ob1k.concurrent.ComposableFutures;
+import com.outbrain.ob1k.concurrent.handlers.FutureSuccessHandler;
+import com.outbrain.swinfra.metrics.api.MetricFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import static com.outbrain.ob1k.concurrent.ComposableFutures.fromError;
+import static com.outbrain.ob1k.concurrent.ComposableFutures.fromNull;
+import static com.outbrain.ob1k.concurrent.ComposableFutures.fromValue;
+
+
+/**
+ * User: aronen
+ * Date: 6/30/13
+ * Time: 6:08 PM
+ * An async cache without a loader cache entries are added only by applying setAsync
+ */
+public class SimpleLocalAsyncCache<K,V> extends AbstractLocalAsyncCache<K,V> {
+  private final Cache<K, ComposableFuture<V>> localCache;
+
+  public SimpleLocalAsyncCache(final int maximumSize, final int ttl, final TimeUnit unit, final MetricFactory metricFactory, final String cacheName) {
+    final boolean collectStats = metricFactory != null;
+    final CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder()
+        .maximumSize(maximumSize)
+        .expireAfterWrite(ttl, unit);
+
+    if (collectStats) {
+      builder.recordStats();
+    }
+
+    this.localCache = builder.build();
+    if (collectStats) {
+      GuavaCacheGaugesFactory.createGauges(metricFactory, localCache, "LocalAsyncCache-" + cacheName);
+    }
+  }
+
+  public SimpleLocalAsyncCache(final int maximumSize, final int ttl, final TimeUnit unit) {
+    this(maximumSize, ttl, unit, null, null);
+  }
+
+  // for testing purposes only!
+  public SimpleLocalAsyncCache() {
+    this(1000, 20, TimeUnit.SECONDS);
+  }
+
+
+  @Override
+  public ComposableFuture<V> getAsync(final K key) {
+    try {
+        final ComposableFuture<V> res = localCache.getIfPresent(key);
+        if (res == null) {
+          return fromNull();
+        }
+        return res;
+    } catch (UncheckedExecutionException | ExecutionError e) {
```

Comment:
at least by the javadoc, no exception at all should be thrown. I think it was just a mistake